lib/jsonutil/jsonutil.go

```diff
@@ -0,0 +1,24 @@
+package jsonutil
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+func SanitizePayload(val interface{}) (interface{}, error) {
+	_, isOk := val.(map[string]interface{})
+	if isOk {
+		return val, nil
+	}
+
+	var jsonMap map[string]interface{}
+	err := json.Unmarshal([]byte(fmt.Sprint(val)), &jsonMap)
```

Comment:
Can we just cast to a string instead of using `fmt.Sprint()`? We can actually just return `val` if it isn't a string in the first place.